### ISSUE TITLE
Tests with --prefer-lowest are failing without roave/security-advisories

### ISSUE THREAD

**Issue by yukoff** [torrentpier/torrentpier#445]
We use codeception for tests, and after merging v2.2.3 they are failing when requirements installed with `composer install --prefer-lowest`. Here is the diff of package versions installed:
```diff
--- tests/travis1.log   2017-08-08 00:43:11.634359500 +0300
+++ tests/travis2.log   2017-08-08 00:42:38.122297500 +0300
@@ -1,6 +1,6 @@
 Loading composer repositories with package information
 Updating dependencies (including require-dev)
-Package operations: 77 installs, 0 updates, 0 removals
+Package operations: 76 installs, 0 updates, 0 removals
   - Installing rych/bencode (v1.0.0): Downloading (100%)
   - Installing symfony/process (v2.3.0): Downloading (100%)
   - Installing gitonomy/gitlib (v1.0.0): Downloading (100%)
@@ -50,8 +50,8 @@
   - Installing guzzlehttp/psr7 (1.0.0): Downloading (100%)
   - Installing react/promise (v2.0.0): Downloading (100%)
   - Installing guzzlehttp/streams (3.0.0): Downloading (100%)
-  - Installing guzzlehttp/ringphp (1.1.0): Downloading (100%)
-  - Installing guzzlehttp/guzzle (5.3.1): Downloading (100%)
+  - Installing guzzlehttp/ringphp (1.0.0): Downloading (100%)
+  - Installing guzzlehttp/guzzle (5.0.0): Downloading (100%)
   - Installing facebook/webdriver (1.0.1): Downloading (100%)
   - Installing behat/gherkin (v4.4.0): Downloading (100%)
   - Installing codeception/codeception (2.3.0): Downloading (100%)
@@ -62,7 +62,7 @@
   - Installing google/recaptcha (1.1.0): Downloading (100%)
   - Installing longman/ip-tools (1.2.0): Downloading (100%)
   - Installing samdark/sitemap (2.0.0): Downloading (100%)
-  - Installing swiftmailer/swiftmailer (v5.4.5): Downloading (100%)
+  - Installing swiftmailer/swiftmailer (v5.4.0): Downloading (100%)
   - Installing symfony/dotenv (v3.3.0): Downloading (100%)
   - Installing symfony/var-dumper (v3.3.0): Downloading (100%)
   - Installing theseer/fdomdocument (1.4.0): Downloading (100%)
@@ -70,10 +70,10 @@
   - Installing sebastian/phpcpd (2.0.0): Downloading (100%)
   - Installing phpoption/phpoption (1.0.0): Downloading (100%)
   - Installing doctrine/lexer (v1.0): Downloading (100%)
-  - Installing doctrine/annotations (v1.2.7): Downloading (100%)
+  - Installing doctrine/annotations (v1.0): Downloading (100%)
   - Installing phpcollection/phpcollection (0.1.0): Downloading (100%)
   - Installing jms/parser-lib (1.0.0): Downloading (100%)
   - Installing jms/metadata (1.1.1): Downloading (100%)
   - Installing jms/serializer (1.0.0): Downloading (100%)
   - Installing scrutinizer/ocular (1.3.0): Downloading (100%)
-  - Installing squizlabs/php_codesniffer (2.8.1): Downloading (100%)
+  - Installing squizlabs/php_codesniffer (2.7.0): Downloading (100%)
```
here `travis1.log` is successful one, `travis2.log` is after merging v2.2.3

Here is the actual error:
```
> vendor/bin/codecept run --steps --coverage --coverage-xml=clover.xml
Codeception PHP Testing Framework v2.3.0
Powered by PHPUnit 5.4.3 by Sebastian Bergmann and contributors.

Fatal error: Redefinition of parameter $_ in /var/www/html/vendor/guzzlehttp/ringphp/src/Client/StreamHandler.php on line 303

Call Stack:
    0.0003     356944   1. {main}() /var/www/html/vendor/codeception/codeception/codecept:0
    0.0806    2152376   2. Codeception\Application->run() /var/www/html/vendor/codeception/codeception/codecept:36
    0.0806    2152376   3. Codeception\Application->run() /var/www/html/vendor/codeception/codeception/src/Codeception/Application.php:103
    0.0835    2230368   4. Codeception\Application->doRun() /var/www/html/vendor/symfony/console/Application.php:126
    0.0837    2230368   5. Codeception\Application->doRunCommand() /var/www/html/vendor/symfony/console/Application.php:195
    0.0837    2230368   6. Codeception\Command\Run->run() /var/www/html/vendor/symfony/console/Application.php:878
    0.0841    2233648   7. Codeception\Command\Run->execute() /var/www/html/vendor/symfony/console/Command/Command.php:259
    0.1069    2552512   8. Codeception\Command\Run->runSuites() /var/www/html/vendor/codeception/codeception/src/Codeception/Command/Run.php:303
    0.1069    2552512   9. Codeception\Codecept->run() /var/www/html/vendor/codeception/codeception/src/Codeception/Command/Run.php:376
    0.1081    2557280  10. Codeception\Codecept->runSuite() /var/www/html/vendor/codeception/codeception/src/Codeception/Codecept.php:153
    0.1157    2665128  11. Codeception\SuiteManager->initialize() /var/www/html/vendor/codeception/codeception/src/Codeception/Codecept.php:182
    0.1163    2668904  12. Codeception\Module\PhpBrowser->_initialize() /var/www/html/vendor/codeception/codeception/src/Codeception/SuiteManager.php:80
    0.1163    2668904  13. Codeception\Module\PhpBrowser->_initializeSession() /var/www/html/vendor/codeception/codeception/src/Codeception/Module/PhpBrowser.php:132
    0.1201    2745784  14. Codeception\Module\PhpBrowser->_prepareSession() /var/www/html/vendor/codeception/codeception/src/Codeception/Module/PhpBrowser.php:230
    0.1201    2747040  15. GuzzleHttp\Client->__construct() /var/www/html/vendor/codeception/codeception/src/Codeception/Module/PhpBrowser.php:259
    0.1236    2823872  16. GuzzleHttp\Client::getDefaultHandler() /var/www/html/vendor/guzzlehttp/guzzle/src/Client.php:92
    0.1262    2827432  17. spl_autoload_call() /var/www/html/vendor/guzzlehttp/guzzle/src/Client.php:121
    0.1262    2827496  18. Composer\Autoload\ClassLoader->loadClass() /var/www/html/vendor/guzzlehttp/guzzle/src/Client.php:121
    0.1262    2827608  19. Composer\Autoload\includeFile() /var/www/html/vendor/composer/ClassLoader.php:322

Script vendor/bin/codecept run --steps --coverage --coverage-xml=clover.xml handling the test:unit:clover event returned with error code 255
```
So it seems `guzzlehttp/ringphp=1.0.0` is the reason, but it is installed as a dependency and I didn't checked what requires it. According to guzzle/RingPHP#22, this is purely PHP 7 related and has been fixed in RingPHP 1.0.7.

**Comment by Exile37** [comment]
`roave/security-advisories` will back in 2.2.4, thank you.

**Comment by yukoff** [comment]
Created PR #446 just in case.

**Comment by Exile37** [comment]
Fixed #446.
